homeassistant/components/timer/__init__.py

```diff
@@ -69,6 +78,16 @@ def _none_to_empty_dict(value):
         return {}
     return value
 
+def _time_str(time_value):
```

Comment:
Let's not do this. We have a specific way people specify durations, and that should be the way it is done. We should also try not to put too much into a single PR. So let's remove this.

Reply:
If we don't do this, timers > 24 hours will not be restored correctly and they will not display in lovelace. 

Try setting a timer > 24 hours in the current release and you will see what I mean. The countdown doesn't display at all. `str(datetime)` for times > 24 hours returns something like this: `D days, HH:MM:SS` If you have a more elegant way to make it `HH:MM:SS` for times greater than 24 hours, we should do that then. Or if we can update the lovelace code to handle the `D days, HH:MM:SS` format, then I will do that (not sure where the code is that handles that though) and update the restore code to handle it as well. I'm open to any suggestions. Thanks @balloob!

Reply:
Just to throw in my two cents, since I'm the guy who found this bug, it's probably not out of the question that those of us with long timer durations are going to be using this and you should absolutely assume that timers surviving reboots will be >24 hours. Mine are for a week (defined as "168:00:00" in the timer.start duration, but as pointed out being converted to 7 days, 0 hours and causing the lovelace display of the timers to error out) to track chores actually being done.

Reply:
Why are we even using `str(datetime)` to begin with? That's not how you serialize a datetime. You nee to use `datetime_value.total_seconds()`.

Reply:
Did you mean `datetime_value.timestamp()`?

Reply:
Ah, sorry I was thinking of a duration. I should have written `timedelta_value.total_seconds()`. For datetime we should serialize it as `datetime_value.isoformat()`

Reply:
I get using the isoformat, but the End value is included in the timer attributes so that the timer can be restored. Users will see this when they look at the state card. It might be a little confusing/less pretty with a UTC time.

Currently displayed:
`2020-01-25 09:47:39.234619-05:00`
Isoformat:
`2020-01-25T14:47:39.234619`
I thought that seeing your time with your timezone information would be more clear to the user. At first glace you might think that the timer is wrong.

Similarly, if we store the duration as seconds then the Duration attribute will display the number of seconds in the state card in lovelace instead of the pretty `00:00:00` that matches how users are likely to enter this information in the config. Also, I just tested it and the lovelace countdown can't interpret longer than 24 hours, so no countdown is displayed when the timer is active. My `_time_str` worked.

Reply:
I'm going to leave it the way it was and make a new PR for this to allow for timers > 24 hours. We can table the discussion for that PR.